### PR TITLE
[AppBar] Add `defaultTraitCollectionDidChangeBlock` for injecting into AppBars.

### PR DESCRIPTION
**Pull request by romoore**
When AppBars are injected into other view controllers, there should be a mechanism by which clients can respond to trait collections changes in those app bars.

Part of #7849

### CODE REVIEW

File: components/AppBar/src/MDCAppBarNavigationController.h [hunk 1]
```diff
@@ -131,6 +131,14 @@ __attribute__((objc_subclassing_restricted)) @interface MDCAppBarNavigationContr
 - (nullable MDCAppBarViewController *)appBarViewControllerForViewController:
     (nonnull UIViewController *)viewController;
 
+/**
+ A block that is assigned to each assigned to each @c MDCAppBarViewController's
+ @c traitCollectionDidChange:.
+ */
+@property(nonatomic, copy, nullable) void (^defaultTraitCollectionDidChangeBlock)
```
Comment (yarneo) [inline components/AppBar/src/MDCAppBarNavigationController.h]: is there a reason why the block has the default prefix?

File: components/AppBar/tests/unit/MDCAppBarNavigationControllerTests.swift [hunk 2]
```diff
@@ -25,7 +25,7 @@ private class MockAppBarNavigationControllerDelegate:
   }
 }
 
-class AppBarNavigationControllerTests: XCTestCase {
+class MDCAppBarNavigationControllerTests: XCTestCase {
```
Comment (yarneo) [inline components/AppBar/tests/unit/MDCAppBarNavigationControllerTests.swift:28]: is this change intended?

**Review by yarneo — APPROVED** [review]

File: components/AppBar/src/MDCAppBarNavigationController.h (hunk 1 above)
Comment (romoore) [reply components/AppBar/src/MDCAppBarNavigationController.h]: They can be overridden on the `MDCAppBarViewController` itself.  I felt that the "default" conveyed that a bit better.  I'm not sold on the name and would change it if you wanted me to.

File: components/AppBar/tests/unit/MDCAppBarNavigationControllerTests.swift (hunk 2 above)
Comment (romoore) [reply components/AppBar/tests/unit/MDCAppBarNavigationControllerTests.swift:28]: Yeah. Xcode doesn't find associated files (like tests) when they don't match the source file name.  I don't know why we have a habit of mismatching file names, but I fixed this along the way.

File: components/AppBar/src/MDCAppBarNavigationController.h [hunk 3]
```diff
@@ -131,6 +131,14 @@ __attribute__((objc_subclassing_restricted)) @interface MDCAppBarNavigationContr
 - (nullable MDCAppBarViewController *)appBarViewControllerForViewController:
     (nonnull UIViewController *)viewController;
 
+/**
+ A block that is assigned to each assigned to each @c MDCAppBarViewController's
```
Comment (yarneo) [inline components/AppBar/src/MDCAppBarNavigationController.h]: i think there is a bit of a duplication in wording here

File: components/AppBar/src/MDCAppBarNavigationController.h (hunk 1 above)
Comment (yarneo) [reply components/AppBar/src/MDCAppBarNavigationController.h]: oh, are you saying that if we call it just traitCollectionDidChangeBlock it would override the original block?

Comment (romoore) [reply components/AppBar/src/MDCAppBarNavigationController.h]: Sorry, no. I mean that once the `MDCAppBarViewController` is created, the VC it is a child of could potentially override the block.  I was trying to convey the idea that this property isn't affecting this object, but is going to be assigned to the `traitCollectionDidChangeBlock` of the `MDCAppBarViewController` objects that it generates.

I'm pretty sure I've done it poorly (judging by this conversation), so I'm open to suggestions. I know `traitCollectionDidChangeBlock` is a name meant for the object itself (when it receives its own `traitCollectionDidChange:` message).

File: components/AppBar/src/MDCAppBarNavigationController.h (hunk 3 above)
Comment (romoore) [reply components/AppBar/src/MDCAppBarNavigationController.h]: yeah

Comment (romoore) [reply components/AppBar/src/MDCAppBarNavigationController.h]: Done.

File: components/AppBar/src/MDCAppBarNavigationController.h (hunk 1 above)
Comment (yarneo) [reply components/AppBar/src/MDCAppBarNavigationController.h]: ahh got it! What if we say traitCollectionDidChangeBlockForAppBar: or is that too long winded?

Comment (romoore) [reply components/AppBar/src/MDCAppBarNavigationController.h]: There's no such thing as "long-winded" in Objective-C! ;-)

Comment (romoore) [reply components/AppBar/src/MDCAppBarNavigationController.h]: Done.
